chain: add rpc polling events option

ellemouton opened the pull request.

orbitalturtle committed:
chain: add rpc polling events option

In this commit, we add a new implementation of BitcoindEvents that uses
Bitcoind's RPC interface to poll for new blocks and transactions.

orbitalturtle commented on chain/bitcoind_events.go:
```diff
@@ -0,0 +1,523 @@
+					continue
+				}
+
+				b.notifyBlock(newBlock)
```
I'm not seeing this notifyBlock function

ellemouton replied:
it is passed in as a config value. So it is actually the `sendBlockToClients` function

ellemouton committed:
chain: add BitcoindEvents abstraction

In this commit, the zmq subscription members of BitcoindConn are
replaced with a BitcoindEvents abstraction. This will make it easier to
add a different event notification delivery implementation for this in a
future commit.

ellemouton committed:
chain: Tests for bitcoind tx and block events

This commit adds a test that ensures that the expected block and
transaction notifications are received from a bitcoind client during
normal operation and during a reorg.
